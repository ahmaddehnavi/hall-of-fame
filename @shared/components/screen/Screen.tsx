import * as React from 'react';
import {StyleSheet, ViewProps} from 'react-native';
import {Col} from '../layout/Col';

export type ScreenProps = ViewProps & {
    children: React.ReactNode
}

/**
 * can be use for root element of each screen
 */
export class Screen extends React.Component<ScreenProps> {


    render() {
        return (
            <Col
                {...this.props}
                style={[
                    styles.screen,
                    this.props.style
                ]}
            />
        )
    }

}

const styles = StyleSheet.create({
    screen: {
        flex: 1
    }
});