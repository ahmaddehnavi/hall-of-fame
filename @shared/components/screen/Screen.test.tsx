import React from 'react';
import 'react-native';
import {View} from 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Screen} from './Screen';

describe('Screen Tests', () => {

    it('should match snapshot', () => {
        let tree = renderer.create(
            <Screen>
                <View/>
            </Screen>
        );
        expect(tree).toMatchSnapshot();
    });

});
