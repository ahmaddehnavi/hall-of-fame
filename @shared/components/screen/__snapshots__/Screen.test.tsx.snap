// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Tests should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View />
</View>
`;
